File: tools/format.py
#!/usr/bin/env python
# Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
import os
import sys
from third_party import get_prebuilt_tool_path
from util import root_path
from util import run

def main():
    os.chdir(root_path)
    dprint()


def dprint():
    executable_path = get_prebuilt_tool_path("dprint")
    command = [executable_path, "fmt"]
    run(command, shell=False, quiet=True)


if __name__ == "__main__":
    sys.exit(main())

File: tools/format.js
#!/usr/bin/env -S deno run --allow-all --config=tests/config/deno.json
// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
import { join, ROOT_PATH } from "./util.js";

const subcommand = Deno.args.includes("--check") ? "check" : "fmt";
const configFile = join(ROOT_PATH, ".dprint.json");
const cmd = new Deno.Command("deno", {
  args: [
    "run",
    "-A",
    "--no-config",
    "npm:dprint@0.47.2",
    subcommand,
    "--config=" + configFile,
  ],
  cwd: ROOT_PATH,
  stdout: "inherit",
  stderr: "inherit",
});

const { code } = await cmd.output();
Deno.exit(code);
